client/src/App.js

```diff
@@ -17,223 +15,25 @@ import { toIdMap } from "./common/toIdMap";
-				lists,
-				tasks,
-			};
-		};
-
-		const state = await getBoth();
-
-		this.setState({
-			loading: false,
-			// Adding the example data to state!
-			...state,
-		});
-	}
-
-	// LIST FUNCTIONS
-
-	addList(listItem, history) {
-		const oldLists = this.state.lists;
-		const newID = uui();
-
-		const lists = {
-			...oldLists,
-			[newID]: {
-				title: listItem,
-				tasks: [],
-			},
-		};
-
-		this.setState({
-			lists,
-		});
-		localStorage.setItem("lists", JSON.stringify(lists));
-
-		// We want to redirect after state has been set here.
-		// but it needs to be async.
-	}
-
-	deleteList(listKey) {
-		let lists = {
-			...this.state.lists,
-		};
-		let tasks = {
-			...this.state.tasks,
-		};
-
-		const listTasks = lists[listKey].tasks;
-
-		// TODO: Check immutability / best way of doing this?
-		listTasks.map((listTask) => {
-			delete tasks[listTask];
-		});
-		// Delete the list
-		delete lists[listKey];
-
-		this.setState({
-			...this.state.tasks,
-			...this.state.lists,
-			tasks: tasks,
-			lists: lists,
-		});
-		localStorage.setItem("lists", JSON.stringify(lists));
-		localStorage.setItem("tasks", JSON.stringify(tasks));
-	}
-
-	// TASK FUNCTIONS
-
-	// Add Tasks
-
-	addListTask(taskId, newTask) {
-		const randomTaskId = uui();
-
-		const tasks = {
-			...this.state.tasks,
-			[randomTaskId]: {
-				text: newTask,
-				complete: false,
-			},
-		};
-
-		const lists = {
-			...this.state.lists,
-			[taskId]: {
-				...this.state.lists[taskId],
-				tasks: [...this.state.lists[taskId].tasks, randomTaskId],
-			},
-		};
-
-		this.setState({
-			lists,
-			tasks,
-		});
-		localStorage.setItem("lists", JSON.stringify(lists));
-		localStorage.setItem("tasks", JSON.stringify(tasks));
-	}
-
-	// Toogle Task
-
-	toggleTask(taskID) {
-		const tasks = {
-			...this.state.tasks,
-			[taskID]: {
-				...this.state.tasks[taskID],
-				complete: !this.state.tasks[taskID].complete,
-			},
-		};
-		this.setState({
-			...this.state.lists,
-			tasks: tasks,
-		});
-
-		localStorage.setItem("tasks", JSON.stringify(tasks));
-	}
-
-	// Delete task
-
-	deleteTask(listId, taskID) {
-		const updatedTasks = {
-			...this.state.tasks,
-		};
-
-		const updatedLists = {
-			...this.state.lists,
-			[listId]: {
-				...this.state.lists[listId],
-				tasks: this.state.lists[listId].tasks.filter((id) => id !== taskID),
-			},
-		};
-
-		delete updatedTasks[taskID];
-
-		this.setState({
-			lists: updatedLists,
-			tasks: updatedTasks,
-		});
-		// localStorage.setItem('lists', JSON.stringify(lists));
-		// localStorage.setItem('tasks', JSON.stringify(updatedTasks));
-	}
-
 	render() {
-		if (this.state.loading) {
-			return <h1>Loading...</h1>;
-		}
+		// if (this.state.loading) {
+		// 	return <h1>Loading...</h1>;
+		// }
 
-		if (this.state.error) {
-			return <h1>{this.state.error.message || "Something went wrong"}</h1>;
-		}
+		// if (this.state.error) {
+		// 	return <h1>{this.state.error.message || "Something went wrong"}</h1>;
+		// }
```

Comment:
Left commented out for now. We can review error and loading states tomorrow if you want. In short, these should be moved to each page.